packages/mui-system/src/cssVars/cssVarsParser.ts

```diff
@@ -96,21 +98,41 @@ const getCssValue = (keys: string[], value: string | number) => {
  * console.log(css) // { '--fontSize': '12px', '--lineHeight': 1.2, '--palette-primary-500': '#000000' }
  * console.log(vars) // { fontSize: '--fontSize', lineHeight: '--lineHeight', palette: { primary: { 500: 'var(--palette-primary-500)' } } }
  */
-export default function cssVarsParser(theme: Record<string, any>, options?: { prefix?: string }) {
+export default function cssVarsParser(
+  theme: Record<string, any>,
+  options?: {
+    prefix?: string;
+    basePrefix?: string;
+    shouldSkipVar?: (keys: Array<string>, value: string | number) => boolean;
+  },
+) {
   const clonedTheme = { ...theme };
 
   delete clonedTheme.vars; // remove 'vars' from the structure
 
-  const { prefix } = options || {};
+  const { prefix, basePrefix = '', shouldSkipVar } = options || {};
   const css = {} as NestedRecord<string>;
   const vars = {} as NestedRecord<string>;
 
-  walkObjectDeep(clonedTheme, (keys, value) => {
+  walkObjectDeep(clonedTheme, (keys, value, scope) => {
     if (typeof value === 'string' || typeof value === 'number') {
-      const cssVar = `--${prefix ? `${prefix}-` : ''}${keys.join('-')}`;
-      Object.assign(css, { [cssVar]: getCssValue(keys, value) });
+      if (typeof value === 'string' && value.startsWith('var')) {
+        if (prefix) {
+          value = value.replace(basePrefix, prefix);
+        } else {
+          // remove `${basePrefix}-` from var()
+          value = value.replace(`${basePrefix}-`, '');
+        }
+        scope[keys.slice(-1)[0]] = value as string;
```

Comment:
This might deserve a comment of describing its behaviour?

Reply:
Good point.